src/main/java/controller/UserController.java

```diff
@@ -19,22 +19,20 @@ public UserController(UserService userService) {
         this.userService = userService;
     }
 
-    @GetMapping("/get/{id}")
-    public UserResponseDto get(@PathVariable Long id) {
+    @GetMapping("/{id}")
+    public UserResponseDto getUserDto(@PathVariable Long id) {
         User user = userService.get(id);
         UserResponseDto userResponseDto = new UserResponseDto();
         userResponseDto.setEmail(user.getEmail());
         userResponseDto.setName(user.getName());
         return userResponseDto;
     }
 
-    @GetMapping("/getall")
+    @GetMapping
     public List<UserResponseDto> getAll() {
-        List<UserResponseDto> allUsersDto = new ArrayList<>();
-        for (User user : userService.listUsers()) {
-            allUsersDto.add(get(user.getId()));
-        }
-        return allUsersDto;
+        return userService.listUsers().stream()
+                .map(u -> getUserDto(u.getId()))
```

Comment:
you already have users in your stream, but you are passing it's id and fetch each user one more time there.

Reply:
did not quite understand what needs to be done but refactored a little